### PR TITLE
[feat]: [PIE-6571]: Fixing get pipeline and added unit tests, plus fixed sort of input set

### PR DESCRIPTION
**Pull request by archit-harness**
Fixing get pipeline and added unit tests, plus fixed sort of input set

Impact -> All Plan Execution queries as we are moving to projections

Testing: Basic pipelines run manually locally.
For inline

1. pipeline get 
2. pipeline run
3. pipeline execute

Further testing via pre-qa when merged to develop.

ReleaseNotes summary needed? No

Is it new task type? No
Did you make changes to the code in a backward-compatible manner? Yes
Did you add a new class which is added in kryo serialization? - No
Will the new version of the delegate receive old version task and process? Yes
If new version of the pipeline-service receives response for the old task will it be able to handle it successfully? Yes
Did you make a change in delegate task which will be created in pipeline-services via grpc call to manager? No change
If the older version of pipeline-service creates the task parameter and calls newer cg manager, will it be okay? Yes
if the newer version of pipeline-service creates the task parameter and calls older cg manager, will it be okay? Yes
Did you remove an enum → No
Did you add an enum(That too at end of list) → No
FeatureFlag Added/Removed -> No
Verify Changes with affects any -> No

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/40650)

### CODE REVIEW

**Review by brijeshdhakar — APPROVED** [review]
> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40650)* status: :shipit: complete! ✓

File: pipeline-service/service/src/main/java/io/harness/repositories/pipeline/PMSPipelineRepositoryCustomImpl.java [hunk 1]
```diff
@@ -225,13 +225,15 @@ public Optional<PipelineEntity> find(String accountId, String orgIdentifier, Str
...
-      String accountId, String orgIdentifier, String projectIdentifier, String pipelineIdentifier, boolean notDeleted) {
+  private PipelineEntity getPipelineEntity(String accountId, String orgIdentifier, String projectIdentifier,
+      String pipelineIdentifier, boolean notDeleted, boolean metadataOnly) {
     Criteria criteria = PMSPipelineFilterHelper.getCriteriaForFind(
         accountId, orgIdentifier, projectIdentifier, pipelineIdentifier, notDeleted);
     Query query = new Query(criteria);
-    for (String nonMetadataField : PMSPipelineFilterHelper.getPipelineNonMetadataFields()) {
-      query.fields().exclude(nonMetadataField);
+    if (metadataOnly) {
+      for (String nonMetadataField : PMSPipelineFilterHelper.getPipelineNonMetadataFields()) {
```
Comment (namanvermah) [inline pipeline-service/service/src/main/java/io/harness/repositories/pipeline/PMSPipelineRepositoryCustomImpl.java:234]: we need to change the non metadata fields though.

Comment (archit-harness) [reply pipeline-service/service/src/main/java/io/harness/repositories/pipeline/PMSPipelineRepositoryCustomImpl.java:234]: If metadata is true, why do you need yaml?

**Review by namanvermah — APPROVED** [review]
> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40650)* status: :shipit: complete! ✓

**Review by vaibhav-si — APPROVED** [review]
> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40650)* status: :shipit: complete! ✓

**Pull request merged**